torch/optim/nadam.py

```diff
@@ -244,9 +262,19 @@ def _single_tensor_nadam(params: List[Tensor],
         exp_avg_sq = exp_avg_sqs[i]
         mu_product = mu_products[i]
         step_t = state_steps[i]
+
+        # If compiling, the compiler will handle cudagraph checks, see note [torch.compile x capturable]
+        if not torch._utils.is_compiling() and capturable:
+            assert param.is_cuda and mu_product.is_cuda and step_t.is_cuda, \
+                "If capturable=True, params, mu_products, and state_steps must be CUDA tensors."
+
         # update step
         step_t += 1
-        step = _get_value(step_t)
+
+        if capturable:
+            step = step_t
+        else:
+            step = _get_value(step_t)
```

Comment:
Ho, should _get_value handle capturable?

Reply:
One followup from using capturable is that the noncapturable path should not use _get_value or any PT2 related differing code (there's also _dispatch_sqrt). I was thinking of doing that in another PR in case this assumption that the PT2 path should always go in capturable impl does not actually hold. ccing @mlazos who may have better context here!